tests/unit/test_placeholders_with_steps.py

```diff
@@ -387,3 +409,72 @@ def test_schema_validation_for_step_input():
                 "ParamB": "SampleValueB"
             }
         )
+
+def test_step_result_placeholder():
+    step_result = StepResult()
+
+    test_step_01 = Task(
+        state_id='StateOne',
+        result_selector={
+            'ParamA': step_result["foo"],
```

Comment:
nit: Let's use same notation for all keys for consistency - either `"` or `'`

Reply:
addressed in next commit